### PR TITLE
[RAPTOR-5492]deal with ugly list representation in f-string

### PR DESCRIPTION
**Pull request by scottp-ml**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
Get the correct string representation from the elements of a list of enums.

### CODE REVIEW

File: custom_model_runner/datarobot_drum/drum/typeschema_validation.py
Comment (eric-s-s) [inline custom_model_runner/datarobot_drum/drum/typeschema_validation.py:23]: as this is only used in the one function, why not just move it there and get rid of the docs.

also, just return `[str(el) for el in input_list]`

File: custom_model_runner/datarobot_drum/drum/typeschema_validation.py
Comment (eric-s-s) [inline custom_model_runner/datarobot_drum/drum/typeschema_validation.py:263]: this definitely breaks things.  you'll have use your function on L265

**Review by eric-s-s — CHANGES_REQUESTED** [review]
yikes!  see https://engprod.drdev.io/api/pr_redirect/datarobot-user-models/363 for how things got broken.  we no longer PR test the unit tests, because they need to be in `tests/drum/unit` (and i don't know if that will work) because if we did, this would fail.  that PR should have a solution

File: custom_model_runner/datarobot_drum/drum/typeschema_validation.py
Comment (scottp-ml) [reply custom_model_runner/datarobot_drum/drum/typeschema_validation.py:23]: that would still have the [] in the output, which we need to get rid of.

**Comment by eric-s-s** [comment]
@scottp-ml see: https://github.com/datarobot/datarobot-user-models/pull/364  you will want to wait for that and then rebase.

**Comment by eric-s-s** [comment]
also, since you have a very specific want for this error message, please include a test showing that it does what you want.  probably in  `tests.drum.unit.test_typeschema_validation.TestSchemaValidator`  (once the other PR merges.  it's the wrong location here)

File: tests/drum/unit/test_typeschema_validation.py
Comment (eric-s-s) [inline tests/drum/unit/test_typeschema_validation.py:698]: this should really test at a higher level and in the `TestSchemaValidator`

here's a copypaste test i took from: `tests.drum.unit.test_typeschema_validation.TestSchemaValidator.test_data_types`

```
    def test_data_types_error_message(
        self, ten_k_diabetes
    ):
        condition = Conditions.IN
        values = [Values.CAT, Values.NUM]
        target = "readmitted"

        yaml_str = input_requirements_yaml(Fields.DATA_TYPES, condition, values)
        schema_dict = self.yaml_str_to_schema_dict(yaml_str)
        validator = SchemaValidator(schema_dict)
        ten_k_diabetes.drop(target, inplace=True, axis=1)

        match_str = r'has types:( \w+,?){2,3}.* exactly match: CAT, NUM'
        with pytest.raises(DrumSchemaValidationException, match=match_str):
            validator.validate_inputs(ten_k_diabetes)
```

**Review by eric-s-s — APPROVED** [review]
i have nits.  they are opinions and will not hold up this PR one way or the other. otherwise good.

**Review by jmbannon — APPROVED** [review]

**Pull request merged**
